transformers/uws/server.js

```diff
@@ -46,7 +46,12 @@ module.exports = function server() {
       });
     });
 
-    this.service.upgrade(soc._handle.fd, req.headers['sec-websocket-key'], soc.ssl ? soc.ssl._external : null);
+    var ticket = this.service.transfer(soc._handle.fd, soc.ssl ? soc.ssl._external : null);
+    soc.on('close', function(error) {
+      if (!error) {
```

Comment:
You could probably remove the check but I don't see why you should. Socket.destroy() emits 'close' some other tick so there is time for other close's to trigger. Probably they never will but the check documents the behavior. But yeah, you are free to remove it if you want. I use the check in uws.